Get all submissions fails on Windows

There seem to have some issues with the subprocess library.

when running "make get-all-submissions", error occurs while using start method of SyncManager()

error message

python scripts/leetcode_tools.py get-all-submissions
("Can't pickle local object 'get_all_submissions.<locals>.mgr_init'",)
Traceback (most recent call last):
  File "scripts/leetcode_tools.py", line 197, in <module>
    clize.run(get_question, submit_question, leetcode_login, get_all_submissions, remove_question)
  File "C:\Users\trian\.conda\envs\leetcode\lib\site-packages\sigtools\modifiers.py", line 158, in __call__
    return self.func(*args, **kwargs)
  File "C:\Users\trian\.conda\envs\leetcode\lib\site-packages\clize\runner.py", line 363, in run
    ret = cli(*args)
  File "C:\Users\trian\.conda\envs\leetcode\lib\site-packages\clize\runner.py", line 220, in __call__
    return func(*posargs, **kwargs)
  File "C:\Users\trian\.conda\envs\leetcode\lib\site-packages\clize\runner.py", line 262, in _cli
    return func('{0} {1}'.format(name, command), *args)
  File "C:\Users\trian\.conda\envs\leetcode\lib\site-packages\clize\runner.py", line 220, in __call__
    return func(*posargs, **kwargs)
  File "scripts/leetcode_tools.py", line 170, in get_all_submissions
    manager.shutdown()
AttributeError: 'SyncManager' object has no attribute 'shutdown'
make: *** [Makefile:6: get-all-submissions] Error 1
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\trian\.conda\envs\leetcode\lib\multiprocessing\spawn.py", line 107, in spawn_main
    new_handle = reduction.duplicate(pipe_handle,
  File "C:\Users\trian\.conda\envs\leetcode\lib\multiprocessing\reduction.py", line 79, in duplicate
    return _winapi.DuplicateHandle(
PermissionError: [WinError 5] 액세스가 거부되었습니다

and error happens in the line below

https://github.com/yurirocha15/leetcode-practice-python/blob/7be075175e566f47d33d76e6065f2b2650b46b3d/scripts/leetcode_tools.py#L115

it seems this stackoverflow link(https://stackoverflow.com/questions/9670926/multiprocessing-on-windows-breaks) is relevant issue. 

It seems to me this error occurs because 'mgr_init' passed to start method is not picklable.

I'm working on it but if you have idea, please tell me.

@yurirocha15

--- a/scripts/leetcode_tools.py
+++ b/scripts/leetcode_tools.py
@@ -177,8 +177,11 @@ def remove_question(id: int):
     qdb.load()
     if qdb.check_if_exists(id):
         data = qdb.get_data()[id]
-        os.remove(data.file_path)
-        os.remove(data.test_file_path)
+        try:
+            os.remove(data.file_path)
+            os.remove(data.test_file_path)
+        except FileNotFoundError as e:
+            print(e.args)
         qdb.delete_question(id)
         qdb.save()
         # update readme

--- a/scripts/leetcode_tools.py
+++ b/scripts/leetcode_tools.py
@@ -11,6 +11,7 @@ from typing import Any, Dict, final
 import clize
 from file_handler import FileHandler, generate_files
 from leetcode_client import LeetcodeClient
+from my_utils import mgr_init
 from question_db import QuestionData, QuestionDB
 from readme_handler import ReadmeHandler
 
@@ -46,7 +47,6 @@ def submit_question(id: int):
     qdb = QuestionDB()
     qdb.load()
     problems = qdb.get_data()
-    print(problems.keys())
     # create submit file
     if qdb.check_if_exists(id):
         file_handler = FileHandler(qdb.get_question(id), _LANGUAGE)
@@ -104,10 +104,6 @@ def get_all_submissions():
     imported_cnt = 0
     slug_to_id_map: Dict[str, int] = {}
 
-    # initializer for SyncManager
-    def mgr_init():
-        signal.signal(signal.SIGINT, signal.SIG_IGN)
-
     try:
         while has_next:
             jobs = []
@@ -149,7 +145,6 @@ def get_all_submissions():
                 p.join()
 
             for data in ret_dict.values():
-                print(data)
                 qdb.add_question(data)
                 imported_cnt += 1
 

--- a/scripts/my_utils.py
+++ b/scripts/my_utils.py
@@ -1,0 +1,6 @@
+import signal
+
+
+# initializer for SyncManager
+def mgr_init():
+    signal.signal(signal.SIGINT, signal.SIG_IGN)

